integration/map.go

```diff
@@ -116,9 +116,34 @@ func RunMapIntegration(ctx context.Context, mapID int64, client trillian.Trillia
 			}
 			leafHash := h.HashLeaf(leaf.LeafValue)
 			if err := merkle.VerifyMapInclusionProof(leaf.Index, leafHash, latestRoot.RootHash, incl.Inclusion, h); err != nil {
-				return fmt.Errorf("inclusion proof failed to verify for key %s: %v", incl.Leaf.Index, err)
+				return fmt.Errorf("VerifyMapInclusionProof(%x): %v", leaf.Index, err)
 			}
 		}
 	}
+	if err := testForNonExistantLeaf(ctx, mapID, client, latestRoot); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Ensure that a query for a leaf that does not exist results in a valid inclusion proof.
+func testForNonExistantLeaf(ctx context.Context, mapID int64,
+	client trillian.TrillianMapClient, latestRoot trillian.SignedMapRoot) error {
+	h := merkle.NewMapHasher(testonly.Hasher)
+	r, err := client.GetLeaves(ctx, &trillian.GetMapLeavesRequest{
+		MapId:    mapID,
+		Revision: latestRoot.MapRevision,
+		Index:    [][]byte{[]byte("doesnotexist....................")},
+	})
+	if err != nil {
+		return err
+	}
+	for _, incl := range r.MapLeafInclusion {
+		leaf := incl.Leaf
+		leafHash := h.HashLeaf(leaf.LeafValue)
+		if err := merkle.VerifyMapInclusionProof(leaf.Index, leafHash, latestRoot.RootHash, incl.Inclusion, h); err != nil {
```

Comment:
I do.  `LeafValue` will be `nil` which makes the inclusion proof a "proof of empty". 
Just added a test for expecting `LeafValue` to be `nil`.